server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -336,6 +332,17 @@ public void stopThreadResourceTracking(long threadId, ResourceStatsType statsTyp
         throw new IllegalStateException("cannot update final values if active thread resource entry is not present");
     }
 
+    /**
+     * Individual tasks can override this if they want to support task resource tracking. We just need to make sure that
+     * the ThreadPool on which the task runs on have runnable wrapper similar to
+     * {@link org.opensearch.common.util.concurrent.OpenSearchExecutors#newAutoQueueFixed}
+     *
+     * @return true if resource tracking is supported by the task
+     */
+    public boolean supportsResourceTracking() {
+        return false;
+    }
+
```

Comment:
For now we're adding support for Search and SearchShard requests which are cancellable tasks. But components are designed in way in which we can extend it to any task which executes something on a ThreadPool. Folks can extend this per use case basis so don't want to limit it to just Cancellable tasks. 

Having `ResourceAwareTask` will cause some inheritance issues. To avoid that resource information was added to task itself in this PR : https://github.com/opensearch-project/OpenSearch/pull/2089